clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -708,6 +745,28 @@ protected Node curNode() {
             return curNode;
         }
 
+        public RetryContext getRetryContext() {
+            return retryContext;
+        }
+
+        public void retry(Call failedCall, Throwable throwable) {
+            long now = time.milliseconds();
+            RetryContext failedCallRetryContext = failedCall.getRetryContext();
+            this.retryContext.update(failedCallRetryContext);
+
+            if (this.retryContext.getTries() > maxRetries) {
```

Comment:
it might be a bit cleaner to invert this if statement to write it like this:

```java
if (this.retryContext.tries() <= maxRetries) {
    log.debug("{} failed: {}. Beginning retry #{}", this, prettyPrintException(throwable), retryContext.getTries());
    runnable.call(this, now);
} else {
    failWithTimeout(now, throwable);
}
```

Reply:
Refactored